### DISCUSSION TITLE
Upgraded skip to 1.1.8 - but still see logging ref to 1.1.1

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#256] [Q&A] [answered]
I'm still not able to re-build my project...  Should be upgraded to 1.1.8

❯ 1.1.8/skip.artifactbundle/macos/skip version
Skip version 1.1.8
❯ pwd
/opt/homebrew/Caskroom/skip

But when I run > skip export and look at the /tmp/**** log files I see:

Error: The skip.tools license was created with a version of the software that expired on 2024-10-01 00:00:00 +0000 and must be re-generated.

Which I was getting BEFORE the  Xcode :  File > Packages > Update...version   - which I thought it worked...  Xcode builds started working.

But  >  skip export fails.

❯ skip version
Skip version 1.1.8
❯ skip upgrade
[✓] Check Skip Updates: 1.1.8
[✓] Skip 1.1.8 is up to date.
❯ pwd
/Users/david/Developer/Fastlane_projects/ppwd
❯ cd ..
❯ skip checkup
[✓] Skip version 1.1.8 (= 1.1.8)
[✓] macOS version 15.0.1 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 6.0 (> 5.9.0)
[✓] Xcode version 16.0 (> 15.0.0)
[✓] Xcode tools SDKs: 8
[✓] Homebrew version 4.4.0 (> 4.1.0)
[✓] Gradle version 8.10 (> 8.6.0)
[✓] Java version 22.0.2 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.1
[✓] Android tools SDKs: 2
[✓] Resolve dependencies (11.12s)
[✓] Build hello-skip (14.64s)
[✓] Test Swift (8.81s)
[✓] Test Kotlin (48.3s)
[✓] Archive iOS ipa (20.8s)
[✓] Assemble HelloSkip-release.ipa 40 KB
[✓] Verify HelloSkip-release.ipa 40 KB
[✓] Assembling Android apk (135.31s)
[✓] Verify HelloSkip-release.apk 13.9 MB
[✓] Check Swift Package (1.72s)
[✓] Check Skip Updates: 1.1.8
[✓] Skip 1.1.8 checkup succeeded in 242.61s

❯ skip export
[✓] Check Swift Package (1.32s)
[✗] Build project ppwd (15.14s)
[✓] Archive iOS ipa (27.5s)
[✓] Assemble painlesspassword-debug.ipa 1.8 MB
[✓] Archive painlesspassword-debug.xcarchive.zip 4.1 MB
[✓] Archive iOS ipa (21.92s)
[✓] Assemble painlesspassword-release.ipa 962 KB
[✓] Archive painlesspassword-release.xcarchive.zip 4.5 MB
[✗] Assemble Android app painlesspassword (8.06s)
[✗] Export painlesspassword-debug.apk: The file “painlesspassword-debug.apk” couldn’t be opened because there is no such file.
[✗] Export painlesspassword-release.apk: The file “painlesspassword-release.apk” couldn’t be opened because there is no such file.
[✗] Bundle Android app painlesspassword (1.22s)
[✗] Export painlesspassword-debug.aab: The file “painlesspassword-debug.aab” couldn’t be opened because there is no such file.
[✗] Export painlesspassword-release.aab: The file “painlesspassword-release.aab” couldn’t be opened because there is no such file.
[✓] Export project for painlesspassword
[✓] Archive project source painlesspassword-project.zip 184 bytes
[✓] Skip export ppwd to ./.build/skip-export (78.71s)
[✗] Skip 1.1.8 export failed with 7 errors
Output logged to: /tmp/skip-export-2024-10-08T01:24:07Z.txt
See https://skip.tools/docs/faq and https://community.skip.tools for help
Error: 8 errors
❯ open /tmp/skip-export-2024-10-08T01:24:07Z.txt

Find ref to skip 1.1.1  and errors with license file...
note: Skip 1.1.1: transpile plugin to: /Users/david/Developer/Fastlane_projects/ppwd/.build/plugins/outputs/ppwd/painlesspassword/destination/skipstone/painlesspassword/src/main at 19:24:24
error: An error occurred while performing transpilation: The skip.tools license was created with a version of the software that expired on 2024-10-01 00:00:00 +0000 and must be re-generated.
Error: The skip.tools license was created with a version of the software that expired on 2024-10-01 00:00:00 +0000 and must be re-generated.

[skip-export-2024-10-08T01:24:07Z.txt](https://github.com/user-attachments/files/17286246/skip-export-2024-10-08T01.24.07Z.txt)

Seems WEIRD that I've got 1.1.8 but the log file does NOT - it's 1.1.1  - how do I debug this?

**Comment by davidakoontz** [comment]
Here's something WEIRD...

after deleting the .build folder several times - and rebuilding...

❯ pwd
/Users/david/Developer/Fastlane_projects/ppwd/.build/artifacts/skip/skip/skip.artifactbundle/macos
❯ ./skip version
Skip version 1.1.1

Down inside that .build folder so thing copies in the skip command and it is version 1.1.1  Not 1.1.8 !!!

So some where in the Gradle build there exist a cached 1.1.1 Skip tools that is being copied into the projects/.build/artifacts....

Riddle me ... this one Batman...

**Reply by aabewhite** [reply]
Hmm... is this from Xcode or the command line? Can you try deleting Package.resolved and try again?

**Reply by davidakoontz** [reply]
this was command line...  

>skip export

deleted Package.resolved and reran  skip export.

Same 7 errors

**Reply by davidakoontz** [reply]
should I have done a Xcode File Package Update?

I've no idea what the Package.resolved is a signal flag...???

**Reply by davidakoontz** [reply]
[PackageResolve.pdf](https://github.com/user-attachments/files/17286914/PackageResolve.pdf)

Here's a PDF of Package.resolve  - I noticed version 1.1.1 in it.

**Reply by aabewhite** [reply]
That's the new Package.resolved or the old? Can you run "swift package update" and see if it updates to the latest versions? Or is you Package.swift file pinning to certain versions?

**Reply by davidakoontz** [reply]
That's the newly created Package.resolved file - I guess the result of  a  > skip export   command.

The Package.swift file pinning???  Don't know the meaning of that... but the XML starts with "pins"

<img width="651" alt="Screenshot 2024-10-07 at 9 52 04 PM" src="https://github.com/user-attachments/assets/0c0babaa-d73f-47e3-ab38-ac46a703fcb4">

**Comment by davidakoontz** [comment]
<img width="419" alt="Screenshot 2024-10-07 at 9 21 41 PM" src="https://github.com/user-attachments/assets/27199365-654e-4cd6-bd38-43c9a3aa145b">

Xcode shows skip 1.1.8 

Package.resolve deleted and reRan  Xcode  File > Packages > Update...

Then > skip export  - command line

No luck there ... looking in Package.resove  (AFTER deleting it )  see Skip ver 1.1.1 !!!  Still.

{
  "pins" : [
    {
      "identity" : "skip",
      "kind" : "remoteSourceControl",
      "location" : "https://source.skip.tools/skip.git",
      "state" : {
        "revision" : "620193d5edefdd30ae742ed916a1aa3d8f6b6f52",
        "version" : "1.1.1"
      }
    },

**Reply by davidakoontz** [reply]
Here is Package.swift - noticed a ref to 1.1.1 in it line 17.

❯ vi Package.swift


  1 // swift-tools-version: 5.9
  2 // This is a Skip (https://skip.tools) package,
  3 // containing a Swift Package Manager project
  4 // that will use the Skip build plugin to transpile the
  5 // Swift Package, Sources, and Tests into an
  6 // Android Gradle Project with Kotlin sources and JUnit tests.
  7 import PackageDescription
  8 
  9 let package = Package(
 10     name: "ppwd",
 11     defaultLocalization: "en",
 12     platforms: [.iOS(.v16), .macOS(.v13), .tvOS(.v16), .watchOS(.v9), .macCatalyst(.v16)],
 13     products: [
 14         .library(name: "painlesspasswordApp", type: .dynamic, targets: ["painlesspassword"]),
 15     ],
 16     dependencies: [
 17         .package(url: "https://source.skip.tools/skip.git", from: "1.1.1"),
 18         .package(url: "https://source.skip.tools/skip-ui.git", from: "1.0.0")
 19     ],
 20     targets: [
 21         .target(name: "painlesspassword", dependencies: [.product(name: "SkipUI", package: "skip-ui")], resources: [.process    ("Resources")], plugins: [.plugin(name: "skipstone", package: "skip")]),
 22         .testTarget(name: "painlesspasswordTests", dependencies: ["painlesspassword", .product(name: "SkipTest", package: "s    kip")], resources: [.process("Resources")], plugins: [.plugin(name: "skipstone", package: "skip")]),
 23     ]
 24 )

**Reply by davidakoontz** [reply]
Not knowing what I'm doing but on a lark... updated the version # in Package.swift to:
skip.git from 1.1.8
skip-ui.git from 1.12.0

And builds are looking Good, BETTER... Built!! 


❯ ls
Android          Package.resolved README.md        Sources
Darwin           Package.swift    Skip.env         Tests
❯ rm Package.resolved
❯ ls
Android       Darwin        Package.swift README.md     Skip.env      Sources       Tests
❯ vi Package.swift
❯ skip export
[✓] Check Swift Package (0.75s)
[✓] Build project ppwd (20.34s)
[✓] Archive iOS ipa (14.05s)
[✓] Assemble painlesspassword-debug.ipa 1.8 MB
[✓] Archive painlesspassword-debug.xcarchive.zip 4.1 MB
[✓] Archive iOS ipa (15.68s)
[✓] Assemble painlesspassword-release.ipa 962 KB
[✓] Archive painlesspassword-release.xcarchive.zip 4.5 MB
[✓] Assemble Android app painlesspassword (182.65s)
[✓] Export painlesspassword-debug.apk: 74.3 MB
[✓] Export painlesspassword-release.apk: 15.4 MB
[✓] Bundle Android app painlesspassword (8.4s)
[✓] Export painlesspassword-debug.aab: 19.9 MB
[✓] Export painlesspassword-release.aab: 10.1 MB
[✓] Export project for painlesspassword
[✓] Archive project source painlesspassword-project.zip 1.2 MB
[✓] Skip export ppwd to ./.build/skip-export (245.72s)
[✓] Skip 1.1.8 export succeeded in 245.73s
❯ more Package.swift
// swift-tools-version: 5.9
// This is a Skip (https://skip.tools) package,
// containing a Swift Package Manager project
// that will use the Skip build plugin to transpile the
// Swift Package, Sources, and Tests into an
// Android Gradle Project with Kotlin sources and JUnit tests.
import PackageDescription

let package = Package(
    name: "ppwd",
    defaultLocalization: "en",
    platforms: [.iOS(.v16), .macOS(.v13), .tvOS(.v16), .watchOS(.v9), .macCatalyst(.v16)],
    products: [
        .library(name: "painlesspasswordApp", type: .dynamic, targets: ["painlesspassword"]),
    ],
    dependencies: [
        .package(url: "https://source.skip.tools/skip.git", from: "1.1.8"),
        .package(url: "https://source.skip.tools/skip-ui.git", from: "1.12.0")
    ],
    targets: [
        .target(name: "painlesspassword", dependencies: [.product(name: "SkipUI", package: "skip-ui")], resources: [.process("Resources")], plugins: [.plugin(name: "skipstone", package: "skip")]),
        .testTarget(name: "painlesspasswordTests", dependencies: ["painlesspassword", .product(name: "SkipTest", package: "skip")], resources: [.process("Resources")], plugins: [.plugin(name: "skipstone", package: "skip")]),
    ]
)
❯ more Package.resolved
{
  "pins" : [
    {
      "identity" : "skip",
      "kind" : "remoteSourceControl",
      "location" : "https://source.skip.tools/skip.git",
      "state" : {
        "revision" : "0859b0e274f48a0bf234723be9d419e79b4cb1d4",
        "version" : "1.1.8"
      }
    },
    {
      "identity" : "skip-foundation",
      "kind" : "remoteSourceControl",
      "location" : "https://source.skip.tools/skip-foundation.git",
      "state" : {
        "revision" : "755190df4f0fb1f5c2d5cffca5625a81b57055b5",
        "version" : "1.1.4"
      }
    },
    {
      "identity" : "skip-lib",
      "kind" : "remoteSourceControl",
      "location" : "https://source.skip.tools/skip-lib.git",
      "state" : {
        "revision" : "dc7cf3abd6da8c8813f41cf183449925e2207a04",
        "version" : "1.1.0"
      }
    },
    {
      "identity" : "skip-model",
      "kind" : "remoteSourceControl",
      "location" : "https://source.skip.tools/skip-model.git",
      "state" : {
        "revision" : "d0a9611595d72040c7870261c909679bc08afd1d",
        "version" : "1.2.0"
      }
    },
    {
      "identity" : "skip-ui",


❯

**Reply by davidakoontz** [reply]
That little update of version #s in Package.swift  HELPED a lot.  !!!

Yet the issue is - who/when does Package.swift get updated... and
Why is Package.resolved still lagging behind the Xcode Package versions (see skip-foundation.git  version 1.1.4 but Xcode shows 1.1.6).

I keep saying (to the wind) this tool chain needs a good CLEAN command!

**Reply by marcprux** [reply, marked as answer]
Xcode and SwiftPM's command-line interface don't use the same pins. Xcode's is internal, whereas the CLI uses the `Package.resolved` file. This can indeed be confusing, because Xcode might be using one version of the libraries (after you update using `File`/`Packages`/`Update to latest package version`), whereas command-line `swift` might be using another (which won't be updated until you run `swift package update`, or just delete the `Package.resolved` file).

I agree that this is confusing. We will consider ways to call out when the package may be out of date.

**Reply by davidakoontz** [reply]
Thank you for the explaination - this helps! 

I think I'm coming to a fuller understanding of the issues.  I've got two platforms, one code base, but two build tool chains.  Double the FUN!  Double the disFUNtion :(  especially when I don't understand the workings.

I looked up what PINS could mean...
https://docs.easypost.com/guides/dependency-pinning-guide

Is there a known way to get the two build chains into sync?

**Reply by davidakoontz** [reply]
Thinking that advice for problems that are something like - **delete Package.resolved** are indications of the system of dependencies getting out of sync...  a good old fashioned way of resync-ing was to CLEAN and rebuild...  but there is no CLEAN.  And I'm unsure of where the build process **actually** starts.  Is it build in Xcode?  Or is it skip export?  Or something else?

the more I learn the more I'm amazed that it all works!  At least sometimes!

**Reply by davidakoontz** [reply]
I just reviewed the https://skip.tools/docs/update-guide/  Update Guide...
I didn't see the command  **swift package update** in there... maybe it should be???

<img width="852" alt="Screenshot 2024-10-08 at 9 01 18 AM" src="https://github.com/user-attachments/assets/851cca3b-5b3e-4793-a91f-8b6ba903eedf">

**Reply by davidakoontz** [reply]
That command appears to create / update the Package.resolved file (timestame coordinates ).


But then the Update Guide has us manually editing the Package.swift file to change version #s.

Is this a MANUAL step in the resyncorization of the TWO build chains?  Or is it recreated/updated by some process?

**Comment by davidakoontz** [comment]
Let me point out that after getting builds working again...  the issues I feel contributed to MY confusion and many mistakes I've made...

My ignorance of the two build tool chains.  (A BIG problem)

The need (requirement) to update the two build tool chains.  (I'd not done this since ver 1.1.1 some months back)

The strange fact that some process copies executables that are installed in HomeBrew into the hidden project folder (.build/artifacts/skip...) and uses that absolute path to skip executable to build.  Which is a redundant skip exe that got out of sync with the properly installed skip exe.  This redundancy should be eliminated in the build tool chain.

Oh... and the weird reasoning that there is a HIDDEN (.build) folder in the first place.  Just BAD form for a build system to hide it's work.

The build system does NOT have a proper CLEAN tool/command.  The consumer (ME) does not know which pieces of this are generate/derived files and what is source data.

The build system and tool chain copies in unix paths and then uses the absolute path to execute commands or as parameters (input) to commands - as apposed to using a relative path - or a well known Env Var (oh how I hate the JAVA_HOME Env Var) but it does solve for this issue.

I put this here to help make Skip.tools better in MY future!
